Assets/MixedRealityToolkit.SDK/Experimental/Dialog/Scripts/DialogShell.cs

```diff
@@ -0,0 +1,140 @@
+            foreach (DialogButtonType buttonType in Enum.GetValues(typeof(DialogButtonType)))
+            {
+                if (buttonType == DialogButtonType.None)
+                {
+                    continue;
+                }
+
+                // If this button type flag is set
+                if ((buttonType & result.Buttons) == buttonType)
+                {
+                    buttonTypes.Add(buttonType);
+                }
+            }
+
+            twoButtonSet = new GameObject[2];
+
+            //Find all buttons on dialog...
+            List<DialogButton> buttonsOnDialog = GetAllDialogButtons();
+
+            //set desired buttons active and the rest inactive
+            SetButtonsActiveStates(buttonsOnDialog, buttonTypes.Count);
+
+            //set titles and types
+            if (buttonTypes.Count > 0)
+            {
+                // If we have two buttons then do step 1, else 0
+                int step = buttonTypes.Count == 2 ? 1 : 0;
+                for (int i = 0; i < buttonTypes.Count; ++i)
+                {
+                    twoButtonSet[i] = buttonsOnDialog[i + step].gameObject;
+                    buttonsOnDialog[i + step].SetTitle(buttonTypes[i].ToString());
+                    buttonsOnDialog[i + step].ButtonTypeEnum = buttonTypes[i];
+                }
+            }
+        }
+
+        private void SetButtonsActiveStates(List<DialogButton> buttons, int count)
+        {
+
+            for (int i = 0; i < buttons.Count; ++i)
+            {
+                var flag1 = (count == 1) && (i == 0);
+                var flag2 = (count == 2) && (i > 0);
+                buttons[i].ParentDialog = this;
+                buttons[i].gameObject.SetActive(flag1 || flag2);
+            }
+        }
+
+        private List<DialogButton> GetAllDialogButtons()
+        {
+            List<DialogButton> buttonsOnDialog = new List<DialogButton>();
+            for (int i = 0; i < transform.childCount; i++)
+            {
+                Transform child = transform.GetChild(i);
+                if (child.name == "ButtonParent")
+                {
+                    for (int childIndex = 0; childIndex < child.transform.childCount; ++childIndex)
+                    {
+                        Transform t = child.transform.GetChild(childIndex);
+                        if (t != null)
+                        {
+                            DialogButton button = t.GetComponent<DialogButton>();
+                            if (button != null)
+                            {
+                                buttonsOnDialog.Add(button);
+                            }
+                        }
+                    }
+                }
+            }
+            return buttonsOnDialog;
+        }
+
+        /// <summary>
+        /// Set Title and Text on the Dialog.
+        /// </summary>
+        protected override void SetTitleAndMessage()
+        {
+            var textContentChild = transform.GetChild(0);
```

Comment:
looks like this will override any references I set in the inspector?

Reply:
Shouldn't there be a means of setting the title and message references via inspector or some other means? Hardcoding child gameobjects to search for isn't very robust to change